fastly/client.go

```diff
@@ -317,6 +317,32 @@ func (c *Client) Request(verb, p string, ro *RequestOptions) (*http.Response, er
 	return resp, nil
 }
 
+// parseHealthCheckHeaders returns the serialised body with the custom health
+// check headers appended.
+//
+// NOTE: The Google query library we use for parsing and encoding the provided
+// struct values doesn't support the format `headers=["Foo: Bar"]` and so we
+// have to manually construct this format.
```

Comment:
ℹ️ We had to use the Google library for serialising data because it was the only library that supported a separate form of array syntax our APIs depend on. But unfortunately, the new health check API uses a syntax that even the Google library doesn't support.